Marlin/src/lcd/ultralcd_impl_DOGM.h

```diff
@@ -938,16 +938,31 @@ static void lcd_implementation_status_screen() {
       uint8_t n = LCD_WIDTH - (START_COL) - 1;
       if (longFilename[0]) {
         filename = longFilename;
-        longFilename[n] = '\0'; // cutoff at screen edge
+        #if ENABLED(SCROLL_LONG_FILENAMES)
+          if (isSelected) {
+            uint32_t name_hash = 7;
+            for (uint8_t l = 0; filename[l]; ++l) name_hash = name_hash * 31 + filename[l];
+            if (filename_scroll_hash != name_hash) {                          // If the hash changed...
```

Comment:
The short (8.3) filename is warranted to be different.
How about a simple byte wyse xor?

Reply:
A simple XOR would be fast, but it fails in too simple cases, such as for any anagram. So maybe this would be better…?
```cpp
uint8_t name_hash = 0xAA;
for (uint8_t l = 0; filename[l]; ++l)
  name_hash = ((name_hash << 1) | (name_hash >> 7)) ^ filename[i];
```

Reply:
At first i thought: "Anagram in consecutive files? Should be rare.". But considering "test21" as anagram of "test12", it could be a relative common problem.

Reply:
@thinkyhead: I see. I just tested it on our version of the firmware and now see that the scrolling filename does not work after scrolling unless you backtrack and move to a previous item. Subtle. It probably is more noticeable on the 3 row display where you are scrolling more often than you are in the full graphical display. I like your hash solution.

Reply:
@marcio-ao 
[0 to 3] is 4 rows.

Reply:
@AnHardt: I stand corrected. I've never seen one of those displays in person :)